codec/src/main/java/io/netty/handler/codec/compression/BrotliEncoder.java

```diff
@@ -207,6 +207,7 @@ private void encode(ByteBuf msg, boolean preferDirect) throws Exception {
                 // in only 1 call at `write(ByteBuffer)`.
                 brotliEncoderChannel.write(msg.nioBuffer());
                 brotliEncoderChannel.flush();
+                msg.skipBytes(msg.readableBytes());
```

Comment:
Actually I think this would be more "future-proof". WDYT @vietj @franz1981 
```suggestion
                ByteBuffer nioBuffer = msg.nioBuffer();
                int position = nioBuffer.position();
                brotliEncoderChannel.write(nioBuffer);
                msg.skipBytes(position - nioBuffer.position());
                brotliEncoderChannel.flush();
```

Reply:
mmm it is better indeed, safer

Reply:
it looks better indeed to use this delta than assuming the channel will consume everything